decentraland/unity-renderer
# feat: Add Third Party wearables filter to the backpack

## What does this PR change?
In order to integrate Third Party Wearables, we need to enable them in the backpack. 
For performance reasons, we can't retrieve this information alongside the owned wearables request, and it need to be done individually by each third party by players request.

In this sense, backpack wearables (in all of their categories) need to be filtered by Decentraland or the Third party selected.
Doing so, triggers a call to the catalyst to retrieve the wearables related to that third party exclusively.
Design is pending here.

**Acceptance Criteria**
- [x] Players can select a third party inside the backpack
- [x] The list of available third parties will come from the catalyst
- [x] Selecting a third party retreives the possible wearables related to the player on that specific wallet by triggering a request to the catalyst.
- [x] Third party wearables can be displayed additively (meaning, you see your DCL wearables + the TPW at the same time) or exclusively (either see DCL or the selected TP). This is a UX decision.
- [x] Add a Feature Flag for this feature.

![image](https://user-images.githubusercontent.com/64659061/163810875-3cf576ee-b485-4293-a400-0263f960eeab.png)

**Figma Link:** https://www.figma.com/file/zMetMn5FptTKYXTnxcRtpW/Backpack-Hybrid?node-id=3079%3A3216

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/Add-Third-Party-wearables-filter-to-the-backpack&NETWORK=ropsten
2. Open the Backpack
3. Select different values in the Collections dropdown.
4. Notice the new wearables are appearing.
5. Equip some wearables from these collections and save your profile.
6. Notice your profile keeps the wearables as usual.

## Release notes
**IMPORTANT**: Before release this feature, these things should be merged:
- https://github.com/decentraland/kernel/pull/258
- https://github.com/decentraland/catalyst/issues/1003

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
